cmd/app.go

```diff
@@ -1080,6 +1083,92 @@ func createNamespace(c *cliclient.MasterClient, n string) error {
 	return nil
 }
 
+func processFileInputs(
+	ctx *cli.Context,
+	c *cliclient.MasterClient,
+	tv *managementClient.TemplateVersion,
+	answers map[string]string,
+	values string,
+	interactive bool) (map[string]string, string, error) {
+	var err error
+	if ctx.String("answers") != "" && ctx.String("values") != "" {
+		return nil, "", fmt.Errorf("cannot process an answers file and a values file")
+	}
+	if hasAnswers(ctx.String("answers"), answers) || !hasAnswers(ctx.String("answers"), answers) && !hasValues(ctx.String("values"), values) {
+		values = ""
+		answers, err = processAnswers(ctx, c, tv, answers, interactive)
+		if err != nil {
+			return answers, values, err
+		}
+	}
+	if hasValues(ctx.String("values"), values) {
+		answers = nil
+		values, err = processValues(ctx, tv, values)
+		if err != nil {
+			return answers, values, err
+		}
+	}
+	return answers, values, nil
+}
+
+func hasAnswers(filePathString string, answersMap map[string]string) bool {
+	if filePathString != "" || len(answersMap) != 0 {
+		return true
+	}
+	return false
+}
+
+func hasValues(filePathString string, valuesString string) bool {
+	if filePathString != "" || valuesString != "" {
+		return true
+	}
+	return false
+}
+
+// creates a map of the values file and fills in missing answers with defaults
+func processValues(ctx *cli.Context, tv *managementClient.TemplateVersion, values string) (string, error) {
+	answers := make(map[string]string)
+	if values != "" {
+		//parse values into map to ensure previous values are considered on update
+		defaultsMap, err := createValuesMap([]byte(values))
```

Comment:
I think we shouldn't call this defaults map if we are starting off by copy non-default values to it